include/ginkgo/config.hpp.in

```diff
@@ -67,6 +67,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #cmakedefine GINKGO_MIXED_PRECISION
 
 
+/* Should we compile cuda kernels for Ginkgo? */
+#cmakedefine GINKGO_BUILD_CUDA
+
+
+/* Should we compile hip kernels for Ginkgo? */
+#cmakedefine GINKGO_BUILD_HIP
+
+
```

Comment:
Can this be removed now?

Reply:
I think these are still used ?

Reply:
But not in the public interface, which is where I wanted to avoid configuration-dependent flags as much as possible. So they could be moved to defines on the compilation level.

Reply:
ah, this part still affects the header.
I still use the flag in the devices/cuda/executor or devices/hip/executor.
Do you have any idea?
only one I can imagine is to use compile_option to define this when compiling those cpps requiring the flag.

Reply:
We already have `GKO_COMPILING_CUDA/HIP/OMP`, which is used to select backends for the common kernels and is not defined in hooks.

Reply:
I see.
ginkgo_cuda from cuda has `GKO_COMPILING_CUDA`, but from hook does not have the definition.
ginkgo_cuda_device can use `GINKGO_BUILD_CUDA` to put `GKO_COMPILING_CUDA` or not.
I am not sure whether it is good because cmake contains two place to put the GKO_COMPILING_CUDA (same name but different path). maybe I will use different name first.
we can discuss it more detail later.
For me, switching config.hpp may be still a good way because user can explicit know which flag is on or off.

Reply:
What is the issue with having these flags? What's important is more how we use it than whether it's present here, no? I think users could make use of such flags as well. Sure, mixing and matching Ginkgo headers and libraries will give awkward results for the user, but that's expected and it will not break Ginkgo itself?

Reply:
Yes, they are okay for now. Maybe I should elaborate a bit more on my thoughts about this in one of the next meetings, basically I am trying to make the interface as consistent and independent of the backends as possible.

Reply:
for my understanding, swapping the library can be a behavior selector?
when we use the libginkgo_cuda generated by cuda, we get the cuda function.
when we use the libginkgo_cuda generated by hook, we does not get the cuda function but others can use the same so from the previous without any issue.
I will use another compile flags in devices first and tend to merge it because it will be in the CMake and cpp not public interface such that we can have more flexibility to change it.